### PR TITLE
Remove duplicate access specifiers

### PR DESCRIPTION
**Pull request by sondredl**

### CODE REVIEW

**Comment by timothyqiu** [comment]
Some access specifiers are duplicated on purpose, e.g. for semantic differences.

For example, there are two `public` sections in `TranslationDomain`, one is exposed to the engine and the other is additionally exposed to scripting.

https://github.com/godotengine/godot/blob/87318a2fb7fffeb72adca934e31915be077c3d1f/core/string/translation_domain.h#L67-L74

File: scene/resources/2d/tile_set.h [hunk 1]
```diff
@@ -205,7 +205,6 @@ class TileSet : public Resource {
 	Array compatibility_tilemap_map(int p_tile_id, Vector2i p_coords, bool p_flip_h, bool p_flip_v, bool p_transpose);
 #endif // DISABLE_DEPRECATED
 
-public:
```
Comment (bruvzg) [inline scene/resources/2d/tile_set.h:208]: This one is required, previous `public` is inside `ifdef` and before it, it's `private`.

**Review by bruvzg — COMMENTED** [review]
> Some access specifiers are duplicated on purpose, e.g. for semantic differences.

Not sure if it's the best way to indicate it, and not true for all cases. So we might want to add some custom comment to do it in a more uniform way (if it's desired, I'm not sure about necessity to show it).

**Comment by KoBeWi** [comment]
> For example, there are two public sections in TranslationDomain, one is exposed to the engine and the other is additionally exposed to scripting.

You don't need access specifiers to separate them though, just put a comment.

**Comment by timothyqiu** [comment]
> You don't need access specifiers to separate them though, just put a comment.

A comment becomes ambiguous when there are more than one group in that section.

<details><summary>Example</summary>

```c++
// These are not intended for scripting.
void foo();
void foo_bar();

void foobar();

// Doing things based on bar.
void bar();
void bar_foo();

void whatever();
```

Are `foobar()` and `bar*()` part of "these" described above? Then we invent begin/end pairs in comments.

```c++
// Begin not intended for scripting.

void foo();
void foo_bar();

void foobar();

// Doing things based on bar.
void bar();
void bar_foo();

// End not intended for scripting.

void whatever();
```

They are not that distinguishable from regular comments. Then we invent ASCII arts in comments :laughing: 

</details>

File: core/object/object.h [hunk 2]
```diff
@@ -769,7 +769,6 @@ class Object {
 	static void initialize_class();
 	_FORCE_INLINE_ static void register_custom_data_to_otdb() {}
 
-public:
```
Comment (AThousandShips) [inline core/object/object.h:772]: I'd keep this as the above one has a note and might be changed in the future, and could then cause mistakes

**Comment by adamscott** [comment]
Closing following https://github.com/godotengine/godot/pull/98881#issuecomment-2465101229

**Pull request closed without merging**